### PR TITLE
Integrate the new client with the queue component

### PR DESCRIPTION
**Pull request by jr0me**
|Related issue|
|---|
| https://github.com/wazuh/wazuh-agent/issues/57 |

## Description

This PR makes use of the recently introduced queue in the Communicator.
Some fixes and refactoring were also made.

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux

### CODE REVIEW

File: src/agent/communicator/src/http_client.cpp [hunk 1]
```diff
@@ -129,7 +130,15 @@ namespace http_client
                 continue;
             }
 
-            reqParams.body = messageGetter ? messageGetter() : "";
+            if (messageGetter)
```
Comment (TomasTurina) [inline src/agent/communicator/src/http_client.cpp:133]:
```suggestion
            if (messageGetter != nullptr)
```

File: src/agent/src/agent.cpp [hunk 2]
```diff
@@ -1,6 +1,58 @@
...
+
+                for (const auto& command : jsonObj["commands"])
+                {
+                    messages.emplace_back(Message(MessageType::COMMAND, command));
+                }
+                multiTypeQueue.push(messages);
+            }
+            else
+            {
+                std::cerr << "Invalid JSON structure: 'events' is missing or not an array." << std::endl;
```
Comment (TomasTurina) [inline src/agent/src/agent.cpp:51]:
```suggestion
                std::cerr << "Invalid JSON structure: 'commands' is missing or not an array." << std::endl;
```

File: src/agent/communicator/src/communicator.cpp [hunk 3]
```diff
@@ -101,46 +101,59 @@ namespace communicator
             }();
 
             m_tokenExpTimer->expires_after(duration);
-            co_await m_tokenExpTimer->async_wait(boost::asio::use_awaitable);
+
+            boost::system::error_code ec;
+            co_await m_tokenExpTimer->async_wait(boost::asio::redirect_error(boost::asio::use_awaitable, ec));
+
+            if (ec == boost::asio::error::operation_aborted)
```
Comment (TomasTurina) [inline src/agent/communicator/src/communicator.cpp:108]: Maybe we can put this if inside the next one?

File: src/agent/communicator/src/http_client.cpp (hunk 1 above)
Comment (jr0me) [reply src/agent/communicator/src/http_client.cpp:133]: done

File: src/agent/src/agent.cpp (hunk 2 above)
Comment (jr0me) [reply src/agent/src/agent.cpp:51]: I've removed the error message.

File: src/agent/communicator/src/communicator.cpp (hunk 3 above)
Comment (jr0me) [reply src/agent/communicator/src/communicator.cpp:108]: Yes, I've also removed the `continue`. It wasn't necessary. This is just about printing the correct error message.

File: src/agent/multitype_queue/include/db_path.hpp [hunk 4]
```diff
@@ -0,0 +1,5 @@
+#pragma once
+
+// TODO: should be moved to Config
+constexpr char DEFAULT_FILE_PATH[] = "/home/vagrant/FILE_";
```
Comment (TomasTurina) [inline src/agent/multitype_queue/include/db_path.hpp:4]: Not used.

File: src/agent/multitype_queue/include/db_path.hpp [hunk 5]
```diff
@@ -0,0 +1,5 @@
...
+constexpr char DEFAULT_DB_PATH[] = "queue.db";
```
Comment (TomasTurina) [inline src/agent/multitype_queue/include/db_path.hpp:5]: If this is the only define here, move it to `multitype_queue` file.

File: src/agent/multitype_queue/include/db_path.hpp (hunk 4 above)
Comment (jr0me) [reply src/agent/multitype_queue/include/db_path.hpp]: Removed

File: src/agent/multitype_queue/include/db_path.hpp (hunk 5 above)
Comment (jr0me) [reply src/agent/multitype_queue/include/db_path.hpp:4]: It's also used in the tests.

Comment (jr0me) [reply src/agent/multitype_queue/include/db_path.hpp]: Moved to multitype queue header file.

**Review by TomasTurina — APPROVED** [review]

**Pull request merged**